File: sources/academy/webscraping/scraping_basics_javascript/exercises/imo_countries.mjs
import * as cheerio from 'cheerio';

const url = 'https://www.imo.org/en/ourwork/ero/pages/memberstates.aspx';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const tableElement of $('.content table').toArray()) {
  const $table = $(tableElement);
  const rows = $table.find('tr');

  for (const rowElement of rows.toArray()) {
    const $row = $(rowElement);
    const cells = $row.find('td');

    if (cells.length > 0) {
      const $firstColumn = $(cells[0]);
      const text = $firstColumn.text().trim();
      if (text) {
        console.log(text);
      }
    }
  }

  for (const rowElement of rows.toArray()) {
    const $row = $(rowElement);
    const cells = $row.find('td');

    if (cells.length > 2) {
      const $thirdColumn = $(cells[2]);
      const text = $thirdColumn.text().trim();
      if (text) {
        console.log(text);
      }
    }
  }
}

File: sources/academy/webscraping/scraping_basics_python/exercises/imo_countries.py
import httpx
from bs4 import BeautifulSoup

url = "https://www.imo.org/en/ourwork/ero/pages/memberstates.aspx"

response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for table in soup.select(".content table"):
    for row in table.select("tr"):
        if cells := row.select("td"):
            first_column = cells[0]
            if text := first_column.text.strip():
                print(text)
    for row in table.select("tr"):
        if cells := row.select("td"):
            if len(cells) > 2:
                third_column = cells[2]
                if text := third_column.text.strip():
                    print(text)
